### PR TITLE
Sharing / Likes / Related Posts: do not display markup in ActivityPub requests

### LINKED ISSUE #29435: Sharing: button markup displayed in ActivityPub requests
### Steps to reproduce

1. On a site that's connected to WordPress.com, go to Jetpack > Settings > Sharing and enable the sharing buttons
2. Go to Plugins > Add New, install and activate the ActivityPub plugin
3. View one of your posts on the frontend, and add `/activitypub` to the end of the post URL.
4. You should see a JSON response, with different fields needed for ActivityPub. 2 of those fields are important for us here: `content` and `contentMap`.
5. You will see the markup from the sharing buttons in those fields.

Related issue: #7361

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #29435

## Proposed changes:

See the details in the issue for more information behind that change.

This is fixed using a recent change in the ActivityPub plugin:
https://github.com/Automattic/wordpress-activitypub/pull/328

**Before**

image

**After**

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

> **Note**
> To test, you'll need to be able to make requests to your site. The easiest way to do that is probably via your terminal.

* Start with a site connected to WordPress.com.
* Go to Jetpack > Settings, and activate sharing, likes, and related posts.
* Install and activate the development version of the ActivityPub plugin, available on https://github.com/Automattic/wordpress-activitypub/
* Open a post on the frontend, ensure the sharing buttons as well as the likes are displayed. Related Posts will be displayed too, but only if you have enough published posts on your site for that. Otherwise, you'll only see the Related Posts container div.
* In your terminal, query `curl 'https://yoursite.com/yourpost/' -H 'Accept: application/activity+json'
* Check the output; it should not include the markup from the elements above.

### CODE REVIEW

**Comment by jeherve** [comment]
I've been thinking of potentially better, more fool-proof ways to check for ActivityPub requests. I've opened this PR on the ActivityPub plugin to consider an option there:
https://github.com/Automattic/wordpress-activitypub/pull/328

File: projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php [hunk 1]
```diff
@@ -244,6 +244,14 @@ public function get_headline() {
 	 * @returns string
 	 */
 	public function filter_add_target_to_dom( $content ) {
+		// Do not output Likes on requests for ActivityPub requests.
```
Comment (samiff) [inline projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php:247]:
```suggestion
		// Do not output related posts for ActivityPub requests.
```

File: projects/plugins/jetpack/modules/sharedaddy/sharing-service.php [hunk 2]
```diff
@@ -964,6 +964,14 @@ function sharing_display( $text = '', $echo = false ) {
 		return $text;
 	}
 
+	// Do not output Likes on requests for ActivityPub requests.
```
Comment (samiff) [inline projects/plugins/jetpack/modules/sharedaddy/sharing-service.php]:
```suggestion
	// Do not output sharing buttons for ActivityPub requests.
```

File: projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php (hunk 1 above)
Comment (samiff) [inline projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php]:
```suggestion
		// Do not output related posts for ActivityPub requests.
```

**Review by samiff — COMMENTED** [review]
Tested well for me, just think the two comments need to be updated?

**Comment by jeherve** [comment]
> Tested well for me, just think the two comments need to be updated?

Indeed, bad oversight from me 🤦 thank you!

**Review by samiff — APPROVED** [review]

**Comment by jeherve** [comment]
D116973-code brings this change to WordPress.com Simple.

**Pull request merged**